oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -489,6 +491,11 @@ boolean isConfiguredForDomainWideDelegation() {
     return serviceAccountUser != null && serviceAccountUser.length() > 0;
   }
 
+  private GenericData parseResponseAs(HttpResponse response) throws IOException {
+    GenericData genericData = response.parseAs(GenericData.class);
+    LoggingUtils.logGenericData(genericData, LOGGER, "Auth response payload");
+    return genericData;
+  }
```

Comment:
qq, isn't this the same thing as `LoggingUtils.logGenericData()`? I'm missing why this method needs to exist

Reply:
This `response.parseAs(GenericData.class);` can only be parsed once. So this is a wrapper to parse the data, log and continue with whatever is to do with the genericData.

Reply:
Apologies, I might be missing something obvious but I'm not see where it's parsed more than once and might as be missing why it's different for ServiceAccountCredentials.

What I see is that it's changed to be:
```
GenericData responseData = parseResponseAs(response);
```

but I think it could just be
```
GenericData responseData = response.parseAs(GenericData.class);
LoggingUtils.logGenericData(responseData, LOGGER, "Auth response payload");
...
```
which is the same as it is for the other Credential types (i.e. https://github.com/googleapis/google-auth-library-java/blob/64855267100a73398c9050670655fb578f4d3b59/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java#L300-L302)